e2e/test-runner/runner_test.go

```diff
@@ -16,20 +29,134 @@ func TestBasicSetup(t *testing.T) {
 	}
 
 	// After completing the setup, the OOBE must have created a new user and the distro launcher must have set it as the default.
-	outputStr := tester.AssertWslCommand("whoami")
-	notRoot := strings.Fields(outputStr)[0]
-	if notRoot == "root" {
-		tester.Logf("%s", outputStr)
-		tester.Fatal("Default user should not be root.")
+	{
+		outputStr := tester.AssertWslCommand("whoami")
+		notRoot := strings.Fields(outputStr)[0]
+		if notRoot == "root" {
+			tester.Logf("%s", outputStr)
+			tester.Fatal("Default user should not be root.")
+		}
 	}
 
 	// Subiquity either installs or marks for installation the relevant language packs for the chosen language.
-	outputStr = tester.AssertWslCommand("apt-mark", "showinstall", "language-pack\\*")
-	packs := strings.Fields(outputStr)
-	tester.Logf("%s", outputStr) // I'd like to see what packs were installed in the end, just in case.
-	if len(packs) == 0 {
-		tester.Fatal("At least one language pack should have been installed or marked for installation, but apt-mark command output is empty.")
+	{
+		outputStr := tester.AssertWslCommand("apt-mark", "showinstall", "language-pack\\*")
+		packs := strings.Fields(outputStr)
+		tester.Logf("%s", outputStr) // I'd like to see what packs were installed in the end, just in case.
+		if len(packs) == 0 {
+			tester.Fatal("At least one language pack should have been installed or marked for installation, but apt-mark command output is empty.")
+		}
+	}
+
+	// Proper release. Ensures the right tarball was used as rootfs
+	{
+		expectedRelease := func() string {
+			if *distroName == "Ubuntu-Preview" || *distroName == "UbuntuDev.WslID.Dev" {
+				return "Ubuntu Kinetic Kudu (development branch)"
+			}
+			if *distroName == "Ubuntu" {
+				return "Ubuntu 22.04.1 LTS"
+			}
+			if *distroName == "Ubuntu22.04LTS" {
+				return "Ubuntu 22.04.1 LTS"
+			}
+			if *distroName == "Ubuntu20.04LTS" {
+				return "Ubuntu 20.04.5 LTS"
+			}
+			if *distroName == "Ubuntu18.04LTS" {
+				return "Ubuntu 18.04.6 LTS"
+			}
+			return ""

```

Comment:
This closure requires change whenever a point release of any LTS or whenever the development release gets published or changed into the following cycle. Seems excessive work. We should consider making it more stable, like returning only the series (for example: `18,.04`, `20.04`) and avoid assertions at all if `*distroName == "Ubuntu-Preview" || *distroName == "UbuntuDev.WslID.Dev"`. Or, if we want to be really strict with the release version to the point release, we should consider making it a parameter passed by the CI workflow. What do you think?

Reply:
This test I created as a proposal more than as a done thing. When a new release comes out we always manually check if the correct dot version is downloaded. This semi-automated that. I'm not sure how we could automate it without circular reasoning (e.g: if the pipeline does not have Ubuntu 22.04.2 when time is due, how can we know without manually checking?).

I relaxed the assertion to only check for major and minor without dot release, but I believe it is worth considering.

Reply:
Let's park that idea for the soon future. I think the way to go is having the CI passing which distro version is expected. CI would now by inquiring Launchpad. We already do that when the `build-wsl` workflow decides which versions to build. You'd notice that sometimes it builds more than just the Ubuntu-Preview.  This is a lot more complicated, so I'd rather we investigate that idea in a dedicated PR.